ingestion_server/ingestion_server/indexer.py

```diff
@@ -213,6 +214,30 @@ def _index_table(self, table, dest_idx=None):
             )
             self.replicate(table, dest_idx, query)
 
+    def _bulk_upload(self, es_batch):
+        max_attempts = 4
+        attempts = 0
+        # Initial time to wait between indexing attempts
+        # Grows exponentially
+        cooloff = 5
+        while True:
+            try:
+                deque(helpers.parallel_bulk(self.es, es_batch, chunk_size=400))
+            except elasticsearch.ElasticsearchException:
+                # Something went wrong during indexing.
+                log.warning(
+                    "Elasticsearch rejected bulk query. We will retry in a"
```

Comment:
Is it worth logging the attempt # and how much time will wait until the next attempt?
Are any details about the bulk query logged here, so you can know what query failed (not sure if that would go in `exc_info`)

Reply:
I think you're right that that could be useful. I added attempt number and the cooldown period to the logs.

exc_info will contain detailed information about why the bulk request failed.